File: frontend/recipe-app/src/models/Recipe.ts
export interface Amount {
    value: number;
    unit: string;
}

export type MealType = "BREAKFAST" | "LUNCH" | "DINNER";
  
export interface RecipeIngredient {    
    ingredientId: string;
    quantity: Amount;
}
  
export interface RecipeStep {    
    instruction: string;
    duration: Amount;
}
  
export interface Recipe {
    id: string;
    name: string;
    duration: Amount;
    mealType: MealType[];
    averagePrice: string;
    description: string;
    imageUrl: string;
    categoryIds: string[];
    ingredients: RecipeIngredient[];
    steps: RecipeStep[];
    tenantId: string;
}
  

File: backend/src/main/java/com/example/recipe/entity/Recipe.java
package com.example.recipe.entity;



import com.example.recipe.entity.lookup.Category;
import com.example.recipe.model.*;
import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.ArrayList;
import java.util.List;
import java.util.UUID;


@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
@Document(collection = "recipes")
public class Recipe {
    @Id
    private String id = UUID.randomUUID().toString();
    private String name;

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    private List<MealType> mealTypes = new ArrayList<>();;

    @JsonInclude(JsonInclude.Include.NON_EMPTY)
    private List<FoodOrigin> foodOrigins = new ArrayList<>();;

    private boolean isPublic = false;

    private Amount duration;
    private RelativePrice relativePrice; //= new Amount()
    private String description;
    private String imageUrl;
    private List<String> categoryIds;
    private List<RecipeIngredient> ingredients;
    private List<RecipeStep> steps;
    private String tenantId;
}

